parts/masterparams.t

```diff
@@ -32,6 +32,14 @@
       },
       "type": "string"
     },
+    {{if .MasterProfile.IsVirtualMachineScaleSets}}
+    "agentVnetSubnetID": {
```

Comment:
How is the master profile being VMSS related to the agent subnet? what do you mean by "the agent"? Is it one param for all agentpools? does this belong to masterparams? (sorry that's a lot of questions :) )

Reply:
Since Azure VMSS does not allow us to set static private IPs of the specific instances, the private IPs have to be derived. These IPs are used to join master nodes to the etcd cluster as well as creating the cert and key pairs for etcd. Hence we need to keep the master nodes and the agent nodes in different subnets to ensure all masters are created sequentially. To create separate subnets for master and agent pool nodes, we now have `masterSubnet` and `agentSubnet` in our VNET and our master NSG. Similarly when users have customvnet, we also need to know the agent pool nodes subnet from `agentVnetSubnetID`.